src/App.tsx

```diff
@@ -2,16 +2,42 @@ import { useState } from "react";
 import { Route, RouterProvider, createBrowserRouter, createRoutesFromElements } from "react-router-dom";
 import "./App.css";
 import SplashScreen from "./components/SplashScreen";
+import FormTransition from "./pages/FormTransition";
 import Layout from "./pages/Layout";
-import PaniniCreatorLayout from "./pages/PaniniCreatorLayout";
+import PaniniCreator from "./pages/PaniniCreator";
+
+// 0 = start, restarted state, panini, restart
+export type UserStep = 0 | 1 | 2 | 3;
+export type SetOrderData = (val: object) => void;
+export type UpdateUserStep = () => void;
+
 function App() {
-  const [orderData, setOrderData] = useState({});
+  const [orderData, setOrderData] = useState<object>({});
+  const [userStep, setUserStep] = useState<UserStep>(0);
+  console.log(userStep);
+  const handleUserStepUpdate: UpdateUserStep = () => {
```

Comment:
**nitpick:** Why do you need it for? You have router implemented already - it complicates codebase pretty well

Reply:
User step is used for handling form transition. Because i had a problem to make an easy solution to wrap my head arount this, i've decided to simplify this with steps, since the user navigation through app had 3 steps (1 with 2 variants) hence, i've decided to implement it this way.  But tbh now i can see an easier way with success route and beginning .- .